### PR TITLE
Indirect Data Analysis - post execution plot and save

### LINKED ISSUE #17022: Indirect Data Analysis - Post execution plotting and saving
Indirect Interfaces should be able to plot and save after the algorithm is run by clicking a plot and save button rather than having to specify plot and save options before the interface is executed.

This issue should cover the post plotting and saving implementation for the `Indirect > Data Analysis` interface

* [x] Elwin
* [x] MSD Fit
* [x] I(Q, t)
* [x] I(Q, t) Fit 
* [x] ConvFit - *completed as example (see #17019)*
* [x] JumpFit

### PR DESCRIPTION
**Pull request by louisemccann**
The plotting and saving of mantid workspaces has been refactored to the interface and push buttons for plotting and saving post algorithm executions have been implemented for all IndirectDataAnalysis tabs.

 **To Test:**
Files to test with: \olympic\babylon5\Public\Louise McCann\DataAnalysis
For each tab load relevant files and run algorithm. Check each plot option plots a suitable graph and that the appropriate workspaces have been saved in your save directory.
- [ ] Elwin
- [ ] MSD Fit
- [ ] I(Q, t)
- [ ] I(Q, t) Fit 
- [ ] JumpFit

Fixes #17022 .

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by raquelalvarezbanos** [comment]
- `Elwin`: `Plot` OK and `Save Result` OK
- `MSD Fit`: unexpected exception when running the reduction. Seems to me that the problem is [this](https://github.com/mantidproject/mantid/pull/17242/files#diff-2ba0e322f03b8ef5c5dccc663a3735aaR91) line. @louisemccann: Does it work for you? or am I doing something really wrong?
- `I(Q, t)`: `Plot` (both single and tiled) OK, `Save` OK
- `I(Q, t) Fit`: both `Plot` and `Save` OK
- `JumpFit`: both `Plot` and `Save` OK

File: MantidQt/CustomInterfaces/src/Indirect/Elwin.cpp [hunk 1]
```diff
@@ -217,22 +218,14 @@ void Elwin::run() {
...
-  }
-
   connect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this,
           SLOT(unGroupInput(bool)));
   m_batchAlgoRunner->executeBatchAsync();
 
   // Set the result workspace for Python script export
   m_pythonExportWsName = qSquaredWorkspace;
+
+  m_elwinAlg = elwinMultAlg;
```
Comment (raquelalvarezbanos) [inline MantidQt/CustomInterfaces/src/Indirect/Elwin.cpp]: Fine, but we could have achieved the same functionality just by renaming `elwinMultAlg` to `m_elwinAlg`, then there's no need to define an additional variable to store the algorithm.

File: MantidQt/CustomInterfaces/src/Indirect/Elwin.cpp [hunk 2]
```diff
@@ -496,6 +493,54 @@ void Elwin::updateRS(QtProperty *prop, double val) {
...
+  checkADSForPlotSaveWorkspace(outputWsSquared, true);
+  plotSpectrum(QString::fromStdString(outputWsSquared));
+
+  auto elfWs = m_elwinAlg->getPropertyValue("OutputELF");
+  checkADSForPlotSaveWorkspace(elfWs, true);
+  plotSpectrum(QString::fromStdString(elfWs), 0, 9);
+
+  if (m_blnManager->value(m_properties["Normalise"])) {
+    auto eltWs = m_elwinAlg->getPropertyValue("OutputELT");
+    checkADSForPlotSaveWorkspace(eltWs, true);
```
Comment (raquelalvarezbanos) [inline MantidQt/CustomInterfaces/src/Indirect/Elwin.cpp]: When setting `Normalise to Lowest Temp` to `True`, I get an error:

```Error while plotting: The workspace "irs26176_graphite002_elwin_elt" could not be found```

as that workspace is not in the ADS.

File: MantidQt/CustomInterfaces/src/Indirect/Elwin.cpp [hunk 3]
```diff
@@ -496,6 +493,54 @@ void Elwin::updateRS(QtProperty *prop, double val) {
...
+  checkADSForPlotSaveWorkspace(outputWsSquared, true);
+  addSaveWorkspaceToQueue(QString::fromStdString(outputWsSquared));
+
+  auto elfWs = m_elwinAlg->getPropertyValue("OutputELF");
+  checkADSForPlotSaveWorkspace(elfWs, true);
+  addSaveWorkspaceToQueue(QString::fromStdString(elfWs));
+
+  if (m_blnManager->value(m_properties["Normalise"])) {
+    auto eltWs = m_elwinAlg->getPropertyValue("OutputELT");
+    checkADSForPlotSaveWorkspace(eltWs, true);
```
Comment (raquelalvarezbanos) [inline MantidQt/CustomInterfaces/src/Indirect/Elwin.cpp]: The same happens here (in addition I think in this case we should be checking `checkADSForPlotSaveWorkspace(eltWs, false)`)

**Comment by raquelalvarezbanos** [comment]
@louisemccann: should these changes be mentioned in the release notes?

**Comment by raquelalvarezbanos** [comment]
Works as expected. @louisemccann Should these changes be mentioned in the release notes?

**Comment by louisemccann** [comment]
@raquelalvarezbanos I thought it might be better to update the release notes generally for all the post plot and save work once I have completed it for all tabs

**Comment by raquelalvarezbanos** [comment]
:shipit:

**Pull request merged**